charts/kyverno/templates/clusterrole.yaml

```diff
@@ -88,142 +22,225 @@ rules:
-  resources:
-  - customresourcedefinitions
-  verbs:
-  - delete
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+    - deletecollection
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
-  name: {{ template "kyverno.fullname" . }}:policycontroller
+  name: {{ template "kyverno.fullname" . }}:view
   labels: {{ include "kyverno.labels" . | nindent 4 }}
     app: kyverno
 rules:
-# background processing, identify all existing resources
 - apiGroups:
   - '*'
   resources:
   - '*'
   verbs:
   - get
   - list
-  - update
   - watch
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
-  name: {{ template "kyverno.fullname" . }}:generatecontroller
+  name: {{ template "kyverno.fullname" . }}:mutate
   labels: {{ include "kyverno.labels" . | nindent 4 }}
     app: kyverno
 rules:
-# process generate rules to generate resources
 - apiGroups:
-  - "*"
+  - ""
   resources:
-  - namespaces
-  - networkpolicies
-  - secrets
   - configmaps
-  - resourcequotas
-  - limitranges
-  {{- range .Values.generatecontrollerExtraResources }}
-  - {{ . }}
-  {{- end }}
+  - events
+  - persistentvolumeclaims
+  - replicationcontrollers
+  - replicationcontrollers/scale
+  - secrets
+  - serviceaccounts
+  - services
+  - services/proxy
   verbs:
-  - create
   - update
-  - delete
-  - list
-  - get
-# dynamic watches on trigger resources for generate rules
-# re-evaluate the policy if the resource is updated
 - apiGroups:
-  - '*'
+  - apps
   resources:
-  - namespaces
+  - daemonsets
+  - deployments
+  - replicasets
+  - statefulsets
   verbs:
-  - watch
-{{- end }}
+  - update
+- apiGroups:
+  - autoscaling
+  resources:
+  - horizontalpodautoscalers
+  verbs:
+  - update
+- apiGroups:
+  - batch
+  resources:
+  - cronjobs
+  - jobs
+  verbs:
+  - update
+- apiGroups:
+  - extensions
+  resources:
+  - daemonsets
+  - deployments
+  - ingresses
+  - networkpolicies
+  - replicasets
+  verbs:
+  - update
+- apiGroups:
+  - policy
+  resources:
+  - poddisruptionbudgets
+  verbs:
+  - update
+- apiGroups:
+  - networking.k8s.io
+  resources:
+  - ingresses
+  - networkpolicies
+  verbs:
+  - update
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - rolebindings
+  - roles
+  verbs:
+  - update
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
-  name: {{ template "kyverno.fullname" . }}:admin-policies
+  name: {{ template "kyverno.fullname" . }}:generate
   labels: {{ include "kyverno.labels" . | nindent 4 }}
-    rbac.authorization.k8s.io/aggregate-to-admin: "true"
     app: kyverno
 rules:
+  - apiGroups:
+      - networking.k8s.io
+    resources:
+      - ingresses
+      - ingressclasses
+      - networkpolicies
+    verbs:
+      - create
+      - update
+      - patch
+      - delete
+  - apiGroups:
+      - ""
+    resources:
+      - namespaces
+      - configmaps
+      - secrets
+    verbs:
```

Comment:
I did some rough scanning here and we seem to lose `get` and `list` for these resources. Is this intended?

Reply:
Yes, the `get` and `list` are covered in the `view` role.

We can repeat here, but it makes it harder to maintain as the roles evolve.